src/builtin/tags/block.ts

```diff
@@ -20,7 +20,7 @@ export default {
     const r = this.liquid.renderer
     const html = childDefined !== undefined
       ? childDefined
-      : yield r.renderTemplates(this.tpls, ctx)
+      : yield r.renderTemplates(this.tpls, ctx, new Emitter(ctx.opts.keepOutputType))
```

Comment:
Change the current `new Emitter()` in `renderTemplates()` (inside its parameter declaration) to creating an emitter based on `ctx.opts.keepOutputType`, so we can avoid sweeping changes on every `renderTemplates()` call (Still, I can see you're missing some calling points).

Reply:
Yes, I struggled to make it work with Typescript. But I jsut realised I could make the emitter param optional, and check if the value is defined at the start of the function, creating the proper emitter depending on the options if it is undefined

Reply:
Adressed in commit 85f30fd9dce27914357b0b547ea1cbd6bd988962